Anteproyecto/README.md

```diff
@@ -1,123 +1,251 @@
+- CH1: Corriente de linea: I(t)
+- CH2: Corriente a inyectar: IC(t)
+
+NOTA: Ambos canales del osciloscopio mide señales de tensión cuya forma de onda es proporcional a la corriente.
+
+De esta manera si se utiliza la función "math" del osciloscopio debería obtenerse una corriente senoidal:
+
+fundamental(t) = I(t) + IC(t)
+
 ### Hardware
-- Nucleo F429ZI
-- Filtro RC antialias
-- Filtro RC para suavizar salida del DAC
-- Sensor de corriente por efecto Hall
-- Detector de ciclos de la línea optoacoplado (3.3V semiciclo (+) y 0v semiciclo (-)) 
 
-### Señal de entrada y salida del detector de ciclos
-La salida está aislada de la entrada mediante un optoacoplador
+#### Diagrama general
+En la *Imagen 5* se muestra el circuito esquemático a utilizar:
+
+![alt text](<img/circuito.png>)
+
+*Imagen 5: Diagrama en bloques general*
+
+#### Circuito de acondicionamiento ACS712
+En la *Imagen 6* se ve el diagrama en bloques del sensor ACS712. Como se observa, la entrada de corriente se mide con un shunt y está aislada de la salida.
+
+![alt text](<img/sens_corr_datasheet.png>)
+
+*Imagen 6: Imagen del sensor y diagrama en bloques del circuito integrado*
+
+Un punto a tener en cuenta es que la señal viene acondicionada entre 0 y 5 V. Por eso hay que re-acondicionarlo para ajustarle el rango dinámico a 0 - 3.3 V.
+
+Por otro lado, hay que añadirle un filtro para evitar aliasing al momento de muestrear en el microcontrolador. En este caso la frecuencia de corte es de 5 kHz, es decir, vamos a leer hasta el armónico 100.
+
+![alt text](<img/acond_sens_corr.png>)
+
+*Imagen 7: Acondicionamiento sensor de corriente*
+
+En la *Imagen 8* e *Imagen 9* se pueden ver las simulaciones en tiempo y en frecuencia para una señal de corriente de 5A.
+
+![alt text](<img/acond_sens_corr_t.png>)
+
+*Imagen 8: Respuesta temporal para entrada de 5A*
+
+![alt text](<img/acond_sens_corr_f.png>)
+
+*Imagen 9: Respuesta en frecuencia para entrada de 5A*
+
+#### Circuito Detector de Ciclos
+Este circuito entrega 3.3 V cuando llega cuando la tensión de línea está en el semi-ciclo positivo y 0 V cuando está en el negativo. En la *Imagen 10* se ve el circuito y en la *Imagen 11* las señales de entrada (línea) y salida (tensión lógica).
+Cabe destacar que la lógica está aislada de la línea mediante un opto-acoplador.
+
+![alt text](<img/detect_ciclos.png>)
+
+*Imagen 10: Circuito de detección de ciclos*
+
+![alt text](<img/detect_ciclos_t.png>)
 
-![alt text](<img/det_ciclos.jpg>)
+*Imagen 11: Formas de onda de entrada (tensión de línea) y salida (tensión lógica)*
 
-### Double buffering en la entrada y la salida
-Tanto la entrada como la salida tendrán un esquema de ping pong buffering, es decir, existirán los punteros
+#### Circuito Display
+Para el display se utiliza el esquema de la *Imagen 12*
+
+![alt text](<img/display.png>)
+
+*Imagen 12: Display*
+
+#### Circuito DAC
+Para el DAC simplemente voy a utilizar un circuito RC para suavizar las altas frecuencias del DAC (*Imagen 13*)
```

Comment:
Nope...
En la teoría del procesamiento de señales en tiempo discreto (Discrete-Time Signal Processing) el nombre esa cosa es "filtro de interpolación", y tiene por misión "fill in the blanks" los espacios entre las deltas de Dyrac que serían la secuencia de salida.
Al haber un hold de primer orden (la señal se queda estable durante el tiempo entre muestras, no "desaparece") aparecen efectos no deseados; pero el filtro se calcula para interpolar.
Entiendo que te referís a la frecuencia de muestreo y pensás en filtrarla lo más posible. Sí, pero no tanto.

- por "circuito del DAC" yo espero ver un DAC... creo que lo tenés integrado, o sea que esto se trata de la última etapa de un sistema de DSP, el filtro de interpolación.

Reply:
Perfecto